### ISSUE TITLE
User ended up with a corrupt token entry in the DB after conflict during upgrade

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#2791]
## Problem Definition

A user contacted us via email and shared some screenshots which gave funny errors when trying to find/insert a token which should have been added via the v1 assets upgrade. The user said they had it already locally but opted to solve the confict by choosing remote.

I got their global.db and it seems that something is wrong.

```
sqlite> SELECT * from assets WHERE identifier="_ceth_0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b";
_ceth_0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b|C|Rally|RLY|||rally-2||0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b

sqlite> SELECT * from ethereum_tokens where address="0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b";
sqlite> 
```
So the asset has an entry in the `assets` table and not in the `ethereum_tokens` table. 

## Task

Figure out how this could have happened and if it's something on our side fix.

**Comment by yabirgb** [comment]
Locally I've tried:

- Exactly the same info local and remote -> DB is ok
- Same address different symbol -> DB is ok
- Tried insert -> delete -> update -> DB is ok
- Also tried from 1.15.X in case something special happened for one version
- Adding account with this asset and try update -> DB is OK
- Adding account with this asset -> delete asset -> update -> Db is OK

From reading the code I can't see a path where an insert of the asset will happen but not for the ethereum token. Need to think more ways of breaking it.

**Comment by LefterisJP** [comment]
Well taking it back. Happened again. We can't brush this off under the rug. We need to figure it out.

**Comment by LefterisJP** [comment]
The second user who saw this said they had those assets manually inserted before. Then those assets are also part of the new upgrade. Which causes the conflict. And at the conflict they chose to keep remote. And this happened.

**Comment by LefterisJP** [comment]
I reproduced this. I know what's wrong. It's going to take some thinking to fix.

**Comment by LefterisJP** [comment]
Should be fixed by #2829
